x-pack/legacy/plugins/siem/public/apps/start_app.tsx

```diff
@@ -17,45 +18,67 @@ import { BehaviorSubject } from 'rxjs';
 import { ErrorToastDispatcher } from '../components/error_toast_dispatcher';
-import { KibanaConfigContext } from '../lib/adapters/framework/kibana_framework_adapter';
+import { compose } from '../lib/compose/kibana_compose';
 import { AppFrontendLibs } from '../lib/lib';
 import { PageRouter } from '../routes';
 import { createStore } from '../store';
 import { GlobalToaster, ManageGlobalToaster } from '../components/toasters';
 import { MlCapabilitiesProvider } from '../components/ml/permissions/ml_capabilities_provider';
+import { useKibanaUiSetting } from '../lib/settings/use_kibana_ui_setting';
 
-export const startApp = async (libs: AppFrontendLibs) => {
+const startApp = (libs: AppFrontendLibs) => {
   const history = createHashHistory();
 
   const libs$ = new BehaviorSubject(libs);
 
   const store = createStore(undefined, libs$.pipe(pluck('apolloClient')));
 
-  libs.framework.render(
-    <EuiErrorBoundary>
-      <I18nContext>
-        <ManageGlobalToaster>
-          <ReduxStoreProvider store={store}>
-            <ApolloProvider client={libs.apolloClient}>
-              <ThemeProvider
-                theme={() => ({
-                  eui: libs.framework.darkMode ? euiDarkVars : euiLightVars,
-                  darkMode: libs.framework.darkMode,
-                })}
-              >
-                <KibanaConfigContext.Provider value={libs.framework}>
+  const AppPluginRoot = memo(() => {
+    const [darkMode] = useKibanaUiSetting(DEFAULT_DARK_MODE);
+    return (
+      <EuiErrorBoundary>
+        <I18nContext>
+          <ManageGlobalToaster>
+            <ReduxStoreProvider store={store}>
+              <ApolloProvider client={libs.apolloClient}>
+                <ThemeProvider
+                  theme={() => ({
+                    eui: darkMode ? euiDarkVars : euiLightVars,
+                    darkMode,
+                  })}
+                >
                   <MlCapabilitiesProvider>
                     <PageRouter history={history} />
                   </MlCapabilitiesProvider>
-                </KibanaConfigContext.Provider>
-              </ThemeProvider>
-              <ErrorToastDispatcher />
-              <GlobalToaster />
-            </ApolloProvider>
-          </ReduxStoreProvider>
-        </ManageGlobalToaster>
-      </I18nContext>
-    </EuiErrorBoundary>
-  );
+                </ThemeProvider>
+                <ErrorToastDispatcher />
+                <GlobalToaster />
+              </ApolloProvider>
+            </ReduxStoreProvider>
+          </ManageGlobalToaster>
+        </I18nContext>
+      </EuiErrorBoundary>
+    );
+  });
+  return <AppPluginRoot />;
+};
+
+const ROOT_ELEMENT_ID = 'react-siem-root';
+
+const App = memo(() => {
+  const libs: AppFrontendLibs = compose();
+  return <div id={ROOT_ELEMENT_ID}>{startApp(libs)}</div>;
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const SiemRootController = ($scope: any, $element: any) => {
```

Comment:
In the previous controller these were [typed](https://github.com/spong/kibana/blob/b92bb8c448821a4b1dcddc9148fa7a56723fc11c/x-pack/legacy/plugins/siem/public/lib/adapters/framework/kibana_framework_adapter.ts#L74) as:

``` ts
controller: ($scope: AppUiKibanaAdapterScope, $element: JQLite) => ({
```

Can we use those types again? No big deal either way as this will all be removed here with the new platform anyway 😅